clients/spark/core/src/main/scala/io/treeverse/clients/LakeFSInputFormat.scala

```diff
@@ -113,21 +188,67 @@ class LakeFSInputFormat extends InputFormat[Array[Byte], WithIdentifier[Entry]]
     val rangesReader: SSTableReader[RangeData] =
       SSTableReader.forMetaRange(job.getConfiguration, metaRangeURL)
     val ranges = read(rangesReader)
-    ranges.map(r =>
+    val res = ranges.map(r =>
       new GravelerSplit(
         r.message,
-        new Path(apiClient.getRangeURL(repoName, new String(r.id)))
+        new Path(apiClient.getRangeURL(repoName, new String(r.id))),
+        r.id,
+        r.message.estimatedSize,        
       )
         // Scala / JRE not strong enough to handle List<FileSplit> as List<InputSplit>;
         // explicitly upcast to generate Seq[InputSplit].
         .asInstanceOf[InputSplit]
     )
+    logger.debug(s"Returning ${res.size} splits")
+    res
   }.asJava
+}
 
-  override def createRecordReader(
-      split: InputSplit,
-      context: TaskAttemptContext
-  ): RecordReader[Array[Byte], WithIdentifier[Entry]] = {
-    new EntryRecordReader(Entry.messageCompanion)
+class LakeFSRepositoryInputFormat extends LakeFSBaseInputFormat {
```

Comment:
AFAICT this reads all metaranges of a repository.  So perhaps LakeFSAllMetarangesInputFormat?  (The current name is quite hard for me, it sounds more like a DF of all the repositories defined on the installation...).

Reply:
Done